packages/xo-web/src/xo-app/dashboard/health/unhealthyVdis.js

```diff
@@ -31,49 +31,67 @@ const COLUMNS = [
     sortCriteria: 'name_label',
   },
   {
-    itemRenderer: (srId, { unhealthyVdiChainsLengthBySr }) => (
+    itemRenderer: (srId, { vdisHealthBySr }) => (
       <div>
-        {map(unhealthyVdiChainsLengthBySr[srId], (chainLength, vdiId) => (
+        {map(vdisHealthBySr[srId].unhealthyVdis, (unhealthyVdiLength, vdiId) => (
           <SingleLineRow key={vdiId}>
             <Col>
               <Vdi id={vdiId} />
             </Col>
             <Col>
-              <span>{_('length', { length: chainLength })}</span>
+              <span>{_('length', { length: unhealthyVdiLength })}</span>
             </Col>
           </SingleLineRow>
         ))}
       </div>
     ),
     name: _('vdisToCoalesce'),
   },
+  {
+    itemRenderer: (srId, { vdisHealthBySr }) => (
+      <div>
+        {map(vdisHealthBySr[srId].vdisWithUnknownVhdParent, vdiId => (
+          <Vdi id={vdiId} />
+        ))}
+      </div>
+    ),
+    name: _('vdisWithInvalidVhdParent'),
+  },
 ]
 
 const UnhealthyVdis = decorate([
   addSubscriptions({
-    unhealthyVdiChainsLengthBySr: subscribeSrsUnhealthyVdiChainsLength,
+    vdisHealthBySr: subscribeSrsUnhealthyVdiChainsLength,
   }),
   provideState({
     computed: {
-      srIds: (state, { unhealthyVdiChainsLengthBySr = {} }) => Object.keys(unhealthyVdiChainsLengthBySr),
+      srIds: (_, { vdisHealthBySr = {} }) => {
+        const srIds = []
+        forEach(vdisHealthBySr, (vdiHealth, srId) => {
+          if (!isEmpty(vdiHealth.unhealthyVdis || vdiHealth.vdisWithUnknownVhdParent.length > 0)) {
```

Comment:
IMHO, `isEmpty`  should checks an object, collection, map, or set  not a `boolean`.